src/utils/optionsdict.cc

```diff
@@ -322,4 +322,32 @@ void OptionsDict::AddSubdictFromString(const std::string& str) {
   parser.ParseMain(this);
 }
 
+void OptionsDict::CheckAllOptionsRead(
+    const std::string& path_from_parent) const {
+  std::string s = path_from_parent.empty() ? "" : path_from_parent + '.';
+  for (auto const& option : TypeDict<bool>::dict_) {
```

Comment:
Maybe add a function to TypeDict to avoid some code duplication?
Something like:

```cpp
void EnsureNoUnusedOptions(const std""string& prefix, const std::string& type_name) {
  for (auto const& option: dict_) {
     // ... ("Unknown " + type_name + " option: " + prefix + option.first);
  }
}
```

And then here just:
```
   TypeDict<int>::EnsureNoUnusedOptions("integer", s);
   // ...
```
?